pkg/exec/exec.go

```diff
@@ -37,67 +37,70 @@ var (
 )
 
 func RunCommandInOperatorPod(ctx context.Context, clientsets *k8sutil.Clientsets, cmd string, args []string, operatorNamespace, clusterNamespace string, exitOnError bool) string {
+	var pod v1.Pod
+	var err error
 
-	pod, err := k8sutil.WaitForPodToRun(ctx, clientsets.Kube, operatorNamespace, "app=rook-ceph-operator")
+	pod, err = k8sutil.WaitForPodToRun(ctx, clientsets.Kube, operatorNamespace, "app=rook-ceph-operator")
 	if err != nil {
 		logging.Fatal(fmt.Errorf("failed to wait for operator pod to run: %v", err))
 	}
 
 	var stdout, stderr bytes.Buffer
 
-	err = execCmdInPod(ctx, clientsets, cmd, pod.Name, "rook-ceph-operator", pod.Namespace, clusterNamespace, args, &stdout, &stderr)
-	if err != nil {
-		logging.Error(err)
-		if exitOnError {
-			os.Exit(1)
-		}
+	execCmdInPod(ctx, clientsets, cmd, pod.Name, "rook-ceph-operator", pod.Namespace, clusterNamespace, args, &stdout, &stderr, exitOnError)
+	if exitOnError {
```

Comment:
If exitOnError, why would we return here? If there was an error it already exited, right? And if there wasn't an error, shouldn't we still continue with line 55?

Reply:
Using exitOnError for the stderr and stdout is confusing me. How about a new flag? `exitOnError` should really just be used for deciding whether to exit immediately when there is an error. Today these two flags might be the same value, but there may be a command in the future that will need different values for these flags.

Reply:
Actually, the name `exitOnError` was confusing. We don't need two variables, I have renamed that to `returnOutput` and the bool value passed, it should be clear now. Let's discuss it today if it is still not clear. Thanks for the reviews.